src/p2lab/pokemon/teams.py

```diff
@@ -18,7 +20,7 @@
 
 class Team:
     def __init__(self, pokemon) -> None:
-        self.pokemon = np.array(pokemon)
+        self.pokemon = np.array(deepcopy(pokemon))
```

Comment:
was not deep-copying causing some hidden state mutation or something?

Reply:
I *think* so, basically when a team was being created it had a situation where 2 Pokemon in a team (or even against each other in the same match) could be affecting each other's state!